### DISCUSSION TITLE
Sponsors-only Repositories Feedback

### DISCUSSION THREAD

**Discussion by jlord** [community/community#10918] [Other Feature Feedback, Questions, & Ideas]
We're excited to launch sponsors-only repositories! Now developers with GitHub Sponsors profiles will be able to attach a private repository to each of their tiers. GitHub will automatically add those sponsors as outside contributors to the repository and keep that list up to date. More details in the [documentation](https://docs.github.com/en/sponsors/receiving-sponsorships-through-github-sponsors/managing-your-sponsorship-tiers#adding-a-repository-to-a-sponsorship-tier).

We know many have been doing this manually and we'd love to hear from you when you use this feature. We'd also love to hear from those trying something like this out with their sponsors for the first time. There's a lot of possibility here: these repositories could be used for sponsorware (access to libraries for just your sponsors) or you could use it to communicate with sponsors through Discussions or Issues. You could also use the repository for early access: sponsors get the first look at what you're building but the repository will be made public later. Have a go with it and let us know what you like and what you still need.

Many thanks, the GitHub Sponsors Team ❤️

**Comment by simonw** [comment]
Having one private repository per tier seems a little bit strange to me. I could see myself wanting to have more than one sponsors-only repository, and being able to attach each of them to more than one tier.

Then I could do things like:

- $5, $10 and $25/month tiers: access to the "next version preview" repo
- $50, $100 and $250/month tiers: access to the "next version preview" repo AND the "priority support issues" repo
- Custom sponsor amount above $50 - same benefits as above

**Reply by schorschii** [reply]
I think so too. The ability to grant sponsors access to multiple repos per tier would be very helpful.

**Comment by MichaelBelgium** [comment]
I love this addition, but I can't select anything of my private repo's. I assume its because of the organization requirement? I wonder why thats a requirement and why normal profile private repo's aren't enough

**Reply by jlord** [reply]
Hi! We started with just using organization-owned repos since they have more granular permissions and sponsors can be added as outside collaborators without write access as is the only option in user-owned repos.

**Reply by StephenHaney** [reply]
Big +1 to allowing personal repos... I set up sponsors specifically for this purpose until I realized I can't select any of my repos in this feature.

**Comment by gorhom** [comment]
Thanks for this feature, i think it is in the right direction to support OSS developers.  I can't wait for `sponsorware` is going to be great too! 

Also, i would like to suggest a new feature: Highlight sponsors created issues on public / private repositories, and create custom notification on their interactions on repositories.

**Comment by larrydiamond** [comment]
This is a great addition - thank you very much!   Much of my work targets a corporate market and I'd like to have an annual support fee managed by Github.   This work you've done is so extremely close to managing this for me.   Can you please add an annual support fee as an option?

**Comment by ljharb** [comment]
It’s very frustrating that i can’t use a user-owned repo as my sponsors-only repo. In general, any time there’s a GitHub feature that applies to orgs and not users, it feels exclusionary and more pro-company than pro-individual, and this time is the same.

I’m aware that permissions aren’t as granular on user-owned repos (for no good reason other than “that’s how they were built”, as far as I’m aware), but that’s a choice i should be able to make. The repo I’ve created for my sponsors is solely for the “discussions” feature; i don’t care that every sponsor will have the same write access - it’s what i want. However, to take advantage of this feature, I’d have to either make a redundant org, or use an existing one - and in either case, the org wouldn’t match the name they’re sponsoring.

**Comment by ljharb** [comment]
Related to https://github.com/github/feedback/discussions/10918#discussioncomment-2104360, it’s pretty concerning that i have the ability to accidentally grant my sponsors access to any private repo i control on any org - including my employer and foundation projects where i happen to be an org owner. Merely having admin on a repo isn’t automatically the same thing as having the implicit permission to add people unilaterally.

Since user-owned repos aren’t permitted, without this escape hatch, users would have no recourse - but it still seems pretty broad and easy to make a mistake. Is there a use case besides “users” for granting access to a repo in “not the sponsored user/org”?

**Comment by DaemonF0rge** [comment]
Yeah I'd like the see the ability to grant access to multiple repos

**Comment by KOLANICH** [comment]
I don't think it is right to transform GitHub into IT-themed OnlyFans. But ... it's your service, it's up to you to decide.

**Comment by retlehs** [comment]
sponsorware is great! but having the ability to only add a single repo per tier means we really can't use this feature the way that we'd like to. we're hoping to grant our sponsors access to several repos.

**Reply by jlord** [reply]
This is great feedback, thanks! It helps up prioritize improvements 👍

**Comment by fboeller** [comment]
Love these sponsor-only repos!

When adding such a repo to a tier, it says: "Organizations that sponsor you are not invited to the repository."
I think in my use case it would be beneficial if all members of a sponsoring organization get access to the repository.

While individual software engineers might want to get access to the repo after reaching it from my blog, they usually work for an organization that would get the actual benefit of this software engineers being able to solve a problem faster with access to my repo.
I think it's less likely that a software engineer working for an org would spend an amount like `5$` on their own if the benefit is for the org and also less likely to ask for the org to pay such a small amount.
Instead, if there would be an option to get access for the full org and it cost something like `1$` per member, then I think that engineer is more likely to ask for the org to pay these `50$` in a 50 engineer company and the org is more likely to fund it since the whole org gets the benefit.

Don't have data to back this up but I think pushing more into the direction of the org paying since the org is getting the benefit would be helpful to increase the amount sponsorhips.

**Comment by fboeller** [comment]
Another use case I have:

On my blog, I'm linking from within different blog post to the sponsor-only GitHub repository (more specifically to branches in it that describe the different state of the code that is described in the blog post).

Readers that aren't sponsors see a 404 page.
It would be great if they instead see some call to action to become a sponsor to access this repo.
Maybe showing the README and About to indicate it's a repo. 
But really anything different from a 404 that allows the user to become a sponsor would already be great.

Reader that are sponsors would then be able to access the specific branches via the links I provide in the blog (already possible) and readers that aren't sponsors click on the link and have the option to become sponsor.
